drivers/i2c/i2c_gecko.c

```diff
@@ -262,3 +262,36 @@ DEVICE_AND_API_INIT(i2c_gecko_1, DT_INST_LABEL(1),
 		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
 		    &i2c_gecko_driver_api);
 #endif /* DT_NODE_HAS_STATUS(DT_DRV_INST(1), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(2), okay)
```

Comment:
Can you look at updating the driver to use an instance macro and DT_INST_FOREACH_STATUS_OKAY?

Reply:
Done. This looks so much nicer :)

I don't have any kits with a SOC old enough to test the path without ```CONFIG_SOC_GECKO_HAS_INDIVIDUAL_PIN_LOCATION```. Also, it looks like none of the boards that use such a SOC have any i2c nodes defined. So testing for that code-path is limited to modifying the DTS for a matching board to include a i2c definition, and ensuring that this compiles for that board, and that it doesn't compile if the location definitions don't match.